### PR TITLE
Dedup receipt data.

### LINKED ISSUE #106: De-dupe receipt IPLDs
In an effort to minimize duplication of data in our IPLD database, we want to

1. No longer publish Receipt IPLDs to the public.blocks table
2. Replace the `mh_key` and `cid` fields in the receipt_cids table with a multihash and cid that instead point to the ReceiptTrie leaf node IPLD that contains, as the value stored in the leaf, the receipt we were previously publishing and referencing directly
3. We need to make it clear that the IPLD we reference is a leaf node and not a standalone receipt, so we should consider renaming the rows to `leaf_mh_key` and `leaf_cid`

### PR DESCRIPTION
**Pull request by arijitAD**
Fixes #106

### CODE REVIEW

File: statediff/indexer/indexer.go [hunk 1]
```diff
@@ -110,7 +110,7 @@ func (sdi *StateDiffIndexer) PushBlock(block *types.Block, receipts types.Receip
 	}
 
 	// Generate the block iplds
-	headerNode, uncleNodes, txNodes, txTrieNodes, rctNodes, rctTrieNodes, logTrieNodes, logLeafNodeCIDs, err := ipld.FromBlockAndReceipts(block, receipts)
+	headerNode, uncleNodes, txNodes, txTrieNodes, rctNodes, rctTrieNodes, logTrieNodes, logLeafNodeCIDs, rctLeafNodeCIDs, err := ipld.FromBlockAndReceipts(block, receipts)
 	if err != nil {
 		return nil, fmt.Errorf("error creating IPLD nodes from block and receipts: %v", err)
 	}
```
Comment (i-norden) [inline statediff/indexer/indexer.go:116]: https://github.com/vulcanize/go-ethereum/blob/38a9e267e1f9cbe6dc6cbaa2981261c875f13c52/statediff/indexer/indexer.go#L118 This statement is off (not introduced here). The number of rctTrie and txTrie nodes should be able to be more than the number of rcts and txs (and always will be assuming there is more than 1 rct or tx). The number of rctTrieLeaf nodes on the otherhand should be the same as the number of rcts.

```go
if len(txNodes) != len(rctNodes) || len(rctNodes) != len(rctLeafNodeCIDs){ // throw error }
```
Are the checks it makes sense to perform here.

**Review by i-norden — APPROVED** [review]
LGTM. One comment, an issue that was already present.

File: statediff/indexer/indexer.go (hunk 1 above)
Comment (arijitAD) [reply statediff/indexer/indexer.go:116]: Done

**Review by ashwinphatak — APPROVED** [review]
watcher-ts smoke tests pass

**Pull request merged**